internal/dns/records.go

```diff
@@ -2,93 +2,104 @@ package dns
 
 import (
 	"context"
+	"slices"
 
 	"github.com/miekg/dns"
 	"github.com/projectdiscovery/dnsx/libs/dnsx"
-)
-
-// Record represents a single DNS record.
-type Record struct {
-	Name  string `json:"name" yaml:"name"`
-	TTL   uint32 `json:"ttl" yaml:"ttl"`
-	Type  string `json:"type" yaml:"type"`
-	Value string `json:"value" yaml:"value"`
-}
 
-// Records represents all of the DNS records for a given domain.
-type Records struct {
-	A     []Record `json:"a" yaml:"a"`
-	AAAA  []Record `json:"aaaa" yaml:"aaaa"`
-	MX    []Record `json:"mx" yaml:"mx"`
-	TXT   []Record `json:"txt" yaml:"txt"`
-	NS    []Record `json:"ns" yaml:"ns"`
-	CNAME []Record `json:"cname" yaml:"cname"`
-}
-
-// RecordsReport represents the report of all DNS records for a given domain including all non-fatal errors that occurred.
-type RecordsReport struct {
-	Domain     string   `json:"domain" yaml:"domain"`
-	DNSRecords Records  `json:"dns_records" yaml:"dns_records"`
-	Errors     []string `json:"errors" yaml:"errors"`
-}
+	osintscan "github.com/Method-Security/osintscan/generated/go"
+)
 
-func getDNSRecords(domain string) (Records, error) {
+func getDNSRecords(domain string, questionTypes []uint16) (osintscan.DnsRecords, error) {
 	options := dnsx.DefaultOptions
-	var questionTypes []uint16 = []uint16{dns.TypeA, dns.TypeAAAA, dns.TypeMX, dns.TypeTXT, dns.TypeNS, dns.TypeCNAME}
 	options.QuestionTypes = questionTypes
 	options.MaxRetries = 5
 	client, err := dnsx.New(options)
 	if err != nil {
-		return Records{}, err
+		return osintscan.DnsRecords{}, err
 	}
 
-	dnsRecords := Records{}
+	dnsRecords := osintscan.DnsRecords{}
 
 	results, err := client.QueryMultiple(domain)
 	if err != nil {
-		return Records{}, err
+		return osintscan.DnsRecords{}, err
 	}
 
-	populateRecords := func(records []string, recordType string) []Record {
-		var dnsRecordsSlice []Record
+	populateRecords := func(records []string, recordType string) []*osintscan.DnsRecord {
+		var dnsRecordsSlice []*osintscan.DnsRecord
 		for _, record := range records {
-			dnsRecord := Record{
+			dnsRecord := osintscan.DnsRecord{
 				Name:  domain,
-				TTL:   results.TTL, // This assumes a common TTL for all records; adjust if needed
+				Ttl:   int(results.TTL), // This assumes a common TTL for all records; adjust if needed
 				Type:  recordType,
 				Value: record,
 			}
-			dnsRecordsSlice = append(dnsRecordsSlice, dnsRecord)
+			dnsRecordsSlice = append(dnsRecordsSlice, &dnsRecord)
 		}
 		return dnsRecordsSlice
 	}
 
-	dnsRecords.A = populateRecords(results.A, "A")
-	dnsRecords.AAAA = populateRecords(results.AAAA, "AAAA")
-	dnsRecords.CNAME = populateRecords(results.CNAME, "CNAME")
-	dnsRecords.MX = populateRecords(results.MX, "MX")
-	dnsRecords.NS = populateRecords(results.NS, "NS")
-	dnsRecords.TXT = populateRecords(results.TXT, "TXT")
+	if slices.Contains(questionTypes, dns.TypeA) {
```

Comment:
will this contains operation ever catch TypeAAAA here and classify it as TypeA (because of the subset)?

Reply:
The `dns.TypeA` value is a `uint16` (same for the other values in the `slices.Contains` calls below), so should be good there